cloudinit/cmd/main.py

```diff
@@ -411,6 +474,9 @@ def main_init(name, args):
     mode = sources.DSMODE_LOCAL if args.local else sources.DSMODE_NETWORK
 
     if mode == sources.DSMODE_NETWORK:
+        if os.path.exists(init.paths.get_runpath(".wait-on-network")):
+            LOG.debug("Will wait for network connectivity before continuing")
+            init.distro.wait_for_network()
```

Comment:
Is there a reason to? `/run` gets cleaned on boot, and if a `cloud-init collect-logs` will contain the file as another piece of evidence. I don't think it would hurt any further invocations of cloud-init on the current boot.

Reply:
The only reason would be later calls on the command line to `cloud-init init` per use-cases like -#5726 not needing to actually shell out to run that wait again.

Reply:
Not a critical case and any calls from the CLI are generally human-driven and not in an early boot path where waiting on network would block at that point in time as it's already configured/up.